tasks/zaharov_g_seidel_int_met/report.md

```diff
@@ -0,0 +1,149 @@
+    new_x = sum / A[i][i]
+    max_diff = max(max_diff, |new_x - x[i]|)
+    x[i] = new_x
+
+  if max_diff < epsilon:
+    break
+```
+
+Time complexity: O(max_iterations × n²) per iteration.
+
+
+## 4. Parallelization Scheme
+
+### MPI Implementation:
+- **Data Distribution**: Matrix rows are distributed cyclically across processes
+- **Communication Pattern**: All-to-all communication using `MPI_Allgatherv` after each iteration
+- **Rank Roles**: All processes compute their assigned rows, then exchange results
+
+**Process distribution**:
+```
+Process 0: rows 0, P, 2P, ...
+Process 1: rows 1, P+1, 2P+1, ...
+...
+Process P-1: rows P-1, 2P-1, 3P-1, ...
+```
+
+**Pseudocode**:
+```python
+# Each process:
+local_rows = assign_rows(rank, total_processes)
+local_A = A[local_rows, :]  # Each process stores only its rows
+local_b = b[local_rows]
+
+while not converged and iterations < max_iter:
+  compute_local_x(local_A, local_b, global_x)
+  MPI_Allgatherv(local_x, global_x)  # Exchange results
+  check_convergence(global_x, prev_x)
+```
+
+
+## 5. Implementation Details
+
+### Code Structure
+```
+tasks/zaharov_g_seidel_int_met/
+├── mpi/
+│   ├── include/ops_mpi.hpp      # MPI interface
+│   └── src/ops_mpi.cpp          # MPI implementation
+├── seq/
+│   ├── include/ops_seq.hpp      # Sequential interface
+│   └── src/ops_seq.cpp          # Sequential implementation
+└── tests/                       # Test suites
+```
+
+**Memory Usage**:
+- Sequential: Stores full n×n matrix (O(n²) memory)
+- MPI: Each process stores only its assigned rows (O(n²/P) memory)
+
+**Assumptions**:
+- Matrix is diagonally dominant (guarantees convergence)
+- System size divisible by number of processes (with remainder handling)
+- All processes participate in computation
+
+
+## 6. Experimental Setup
+
+### Hardware/Software Environment
+- **CPU:** Intel Core i5-10600KF (6 cores, 12 threads, 4.1GHz base)
+- **RAM:** 32GB
+- **OS:** NixOS (Linux-based distribution)
+- **Compiler:** GCC 14.3
+- **Build Type:** Release
+
+### Test Configuration
+- **Matrix size:** 2000×2000
+- **Number of Processes:** 6
+
+## 7. Results and Discussion
+
+### 7.1 Correctness
+Correctness verified through:
+- Residual norm calculation: `||Ax - b|| / n < ε × 1000`
+- Comparison with sequential reference implementation
+- Unit tests covering various system sizes (10, 50, 100)
+- Boundary cases and error conditions
+
+All functional tests pass for both SEQ and MPI implementations with identical results within tolerance.
+
+### 7.2 Performance
+
+#### Execution Times
+Performance results for system size 2000, ε=1e-6, max_iterations=1500:
+
+| Mode     | Implementation | Time, (s) | Speedup |
+|----------|----------------|-----------|---------|
+| Task Run | SEQ            | 0.0094    | 1.00    |
+| Task Run | MPI            | 0.0017    | 5.59    |
+| Pipeline | SEQ            | 0.0405    | 1.00    |
+| Pipeline | MPI            | 0.0268    | 1.51    |
```

Comment:
Please, check on different processes count and write the results down to the report